src/OpenApi/test/OpenApiGeneratorTests.cs

```diff
@@ -771,6 +764,22 @@ public void HandlesEndpointWithMultipleResponses()
 
     }
 
+    [Fact]
+    public void OnlyAddParametersWithCorrectLocations()
+    {
+        var operation = GetOpenApiOperation(([FromBody] int fromBody, [FromRoute] int fromRoute, [FromServices] int fromServices) => { });
+
+        Assert.Single(operation.Parameters);
+    }
+
+    [Fact]
+    public void DoNotAddParametersWithTypeIFormFile()
```

Comment:
You might be able to add the check for empty `operation.Parameters` in `AddsFromFormParameterAsFormFile` instead of adding a new test.